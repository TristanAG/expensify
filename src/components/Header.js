import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>expensify</h1>
    <NavLink to="/" className="link" activeClassName="active-link" exact={true}>dashboard</NavLink>
    <NavLink to="/create" className="link" activeClassName="active-link">create</NavLink>

    <NavLink to="/help" className="link" activeClassName="active-link">help</NavLink>
  </header>
);

export default Header;
