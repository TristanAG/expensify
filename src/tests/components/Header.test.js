import React from 'react';
import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json';
import Header from '../../components/header';

test('should render header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();

  //expect(wrapper.find('h1').text()).toBe('expensify');
  // const renderer = new ReactShallowRenderer()
  // renderer.render(<Header/>);
  //
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  //
  // console.log(renderer.getRenderOutput());
})
