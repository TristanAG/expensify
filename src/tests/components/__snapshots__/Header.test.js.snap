// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    expensify
  </h1>
  <NavLink
    activeClassName="active-link"
    ariaCurrent="true"
    className="link"
    to="/dashboard"
  >
    dashboard
  </NavLink>
  <NavLink
    activeClassName="active-link"
    ariaCurrent="true"
    className="link"
    to="/create"
  >
    create
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
